spec/vshasta_spec.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env sh
+# MIT License
+#
+# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+Describe "vshasta.sh"
+  Include sh/vshasta.sh
+  Context "isgcp valid cases"
+    fixturedir=$(pwd)/vshasta.sh/etc
+    fixturefile="${fixturedir}/google_system"
+
+    Before 'setup'
+    After 'teardown'
+
+    setup() {
+      install -dm 0755 "${fixturedir}" "${fixturedir}"
+      echo > "${fixturefile}"
```

Comment:
I'm not too keen with creating files in tests that aren't in a mktemp and that could show up in git status I think we should never create files in the git checkout as a matter of being sure we never come close to accidentally changing data a developer may be updating in the checkout. Adding to the checkout in the same location will also invite concurrency issues between tests if one test creates a path and another removes it and they are run in parallel which we want to avoid (note the test runner in github actions runs things in parallel just to try to catch that).

Could we instead have this create a fake dir/root via mktemp then point the prefix of the test function to that and the default can just be /? That way we're testing out the /etc/google_system functionality but we just are having that show up in $(mktemp -d sut-XXXXXXXX)/etc/google_system vs /etc/google_system by default.

Also how do we guard against someone firing off isgcp /etc/os_system accidentally as all this is truly testing is a file is present based on path. 

Can we switch this to using dmidecode to look at the system itself for attributes of the bios instead of a file in /etc? I think a more direct positive assertion would be better than testing for a file at all.

Example of what I mean:

```
pit:~ # dmidecode  | grep -i google
        Vendor: Google
        Version: Google
        Manufacturer: Google
        Product Name: Google Compute Engine
        Serial Number: GoogleCloud-45B5F55724C6533D460B1FF89E925980
        Manufacturer: Google
        Product Name: Google Compute Engine
        Serial Number: Board-GoogleCloud-45B5F55724C6533D460B1FF89E925980
        Manufacturer: Google
        Manufacturer: Google
        Manufacturer: Google
        Manufacturer: Google
        Manufacturer: Google
```
We can probably tag off that Google Compute Engine from dmidecode, that should be more portable and won't require any temp files or anything just write a function that wraps something like ^^^ and you can then define the mock function replacing the "real" one. I think thats probably a better option to detect if we're running in gcp than looking for an ancillary file in /etc thinking about this a second time.

Reply:
You had originally requested the file be made that way?  

https://github.com/Cray-HPE/libCSM/pull/12#discussion_r1026671701

Maybe enough time passed where a good reevaluation is in effect.

Valid points though.  I can do some more testing on it.  

There also seems to be a curl option in addition to the dmidecode one:

https://cloud.google.com/compute/docs/instances/detect-compute-engine

Reply:
Using `dmidecode` would avoid the need to do anything in vshastav2 which we're discussing in the vshastav2 dev channel.

Reply:
Ah yeah past me is always an idiot is my only explanation. But thinking about it again it seems probably bad. I think the dmidecode methods probably the safest way as its pretty much the same as for say detecting vmware/qemu/etc....

Reply:
OK, check the latest push.  I mocked some dmidecode and curl stuff.  I can add more cases but wanted to get eyes on it first